### PR TITLE
Feature/add details

### PR DESCRIPTION
**Pull request by JoeyStewart**
What I did:

Added some conditional rendering to containers so that they now display a message if the information is not there. Activities, Amenities, Active Alerts, National Park Designation, and Description are now shown on details page. Activities and Amenities also show an incremental list of up to ten elements, depending on the size of their respective arrays. Minor styling changes throughout application. URL bug is now fixed. Parks with only one picture in media array now display. 

Did this break anything?
- [x] No
- [ ]  Yes
Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x]  Styling
- [x]  Bug fix (non-breaking change which fixes an issue)
- [x]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.
Checklist:
- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [x]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing
What's next?

### CODE REVIEW

File: src/ParkChecklist/ParkChecklist.js [hunk 1]
```diff
@@ -7,222 +7,226 @@ import page from "../resources/page.png";
...
+    "CO",
+    "CT",
+    "DE",
+    "FL",
+    "GA",
+    "HI",
+    "ID",
+    "IL",
+    "IN",
+    "IA",
```
Comment (Jesuitman) [inline src/ParkChecklist/ParkChecklist.js:36]: I hate this array so much haha

File: src/ParkChecklist/ParkChecklist.js [hunk 2]
```diff
@@ -7,222 +7,226 @@ import page from "../resources/page.png";
...
+  ];
 
-    const fetchParkDetails = async (park_code) => {
-        try {
-            const response = await axios.get(`${apiLink}/${park_code}`);
-            const selectedPark = response.data.data;
+  const fetchData = async () => {
+    try {
+      const response = await axios.get(apiLink);
+      setParks(response.data.data);
```
Comment (Jesuitman) [inline src/ParkChecklist/ParkChecklist.js:77]: I like the use of getting rid of our error console in favor of displaying the message to the user, it makes the design better

File: src/ParkChecklist/ParkChecklist.js [hunk 3]
```diff
@@ -7,222 +7,226 @@ import page from "../resources/page.png";
...
+      }
+    });
+  };
 
-    useEffect(() => {
-        fetchData();
-    }, [apiLink, selectedState, showResults, currentPage, navigate]);
+  const fetchParkDetails = async (park_code) => {
+    try {
+      const response = await axios.get(`${apiLink}/${park_code}`);
```
Comment (Jesuitman) [inline src/ParkChecklist/ParkChecklist.js:101]: i love the use of interpolation and imports to make the fetch work, it makes the code super dynamic

File: src/ParkChecklist/ParkChecklist.js [hunk 4]
```diff
@@ -7,222 +7,226 @@ import page from "../resources/page.png";
...
+  const handleStateChange = (event) => {
+    setSelectedState(event.target.value);
+  };
 
-    const handleGoClick = () => {
-        setShowResults(true);
-        navigate(`/Parks/${selectedState}`);
-    };
+  const handleGoClick = () => {
+    setShowResults(true);
```
Comment (Jesuitman) [inline src/ParkChecklist/ParkChecklist.js:122]: was this just a spacing thing? if you did it manually you can instead right click on vs code and click format document to make all the spacing fix itself

File: src/ParkDetails/ParkDetails.css [hunk 5]
```diff
@@ -61,4 +62,22 @@
...
+    background-color: antiquewhite;
+    margin: 10px;
+    text-align: center;
+    border-radius: 10px;
+    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.7);
+    padding-left: 20px;
+    padding-right: 20px; 
+}
+
+.weather-icon {
```
Comment (Jesuitman) [inline src/ParkDetails/ParkDetails.css:79]: love this, it looks so good. thanks for the demo!

File: src/ParkDetails/ParkDetails.js [hunk 6]
```diff
@@ -6,31 +6,73 @@ function ParkDetails({ foundPark }) {
...
   }
-  console.log(foundPark);
+
+  let parkImage;
+  if (foundPark.attributes.media && foundPark.attributes.media.length > 1) {
+    parkImage = (
+      <img
+        className="single-image"
+        src={foundPark.attributes.media[1].url}
+        alt={foundPark.attributes.media[1].altText}
```
Comment (Jesuitman) [inline src/ParkDetails/ParkDetails.js:16]: interesting use of the let variable to allow for the smoother rendering, im glad you were able to fix this bug!

File: src/ParkDetails/ParkDetails.js [hunk 7]
```diff
@@ -6,31 +6,73 @@ function ParkDetails({ foundPark }) {
...
+          <h3>Park Info</h3>
+          <p>{foundPark.attributes.designation}</p>
+          <p>{foundPark.attributes.description}</p>
+        </div>
+      </section>
+      <section className="active-alerts">
+        <div className="single-park-alerts">
+          <h3>Active Alerts</h3>
+          {foundPark.attributes.active_alerts[0] ? (
+            <>
```
Comment (Jesuitman) [inline src/ParkDetails/ParkDetails.js:46]: i am blown away at hopw good this all looks and how well it flows! you knocked it out of the park with this formatting

File: src/SelectedStateParks/SelectedStateParks.css [hunk 8]
```diff
@@ -6,6 +6,20 @@
     margin: 2%;
     margin-bottom: 0px;
  }
+
+.no-state{
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
```
Comment (Jesuitman) [inline src/SelectedStateParks/SelectedStateParks.css:14]: no state is a funny class name

**Review by Jesuitman — APPROVED** [review]

File: src/ParkChecklist/ParkChecklist.js (hunk 4 above)
Comment (JoeyStewart) [reply src/ParkChecklist/ParkChecklist.js:122]: I didn't realize that. The main thing I was messing with here was trying to make a loading screen in-between the parks and park details pages. I did not successfully get that implemented so I took out the modified code and tried to put the original code back. That must be why the formatting is a little off.

**Pull request merged**